scripts/runnable_recipe_ray_vllm_weight_sync.py

```diff
@@ -1628,20 +2276,21 @@ def start_ray(self):
     def _set_metric_logger_to_actors(self):
         self.metric_logger = MetricLoggerActor.remote(self.cfg)
         # Pass the logger handle to each worker
-        for worker in self.rollout_workers:
-            worker.set_metric_logger.remote(self.metric_logger)
+        # for worker in self.rollout_workers:
+        #     worker.set_metric_logger.remote(self.metric_logger)
         for worker in self.ref_workers:
             worker.set_metric_logger.remote(self.metric_logger)
         for worker in self.actor_workers:
             worker.set_metric_logger.remote(self.metric_logger)
 
-    def _create_fsdp_group(self, worker_cls, fsdp_world_size: int):
+    def _create_fsdp_group_and_param_server(self, worker_cls, parameter_server_cls, fsdp_world_size: int, num_vllm_workers: int):
```

Comment:
Let's split this into creating the two components separately. We will just pass env vars and VLLM pointers to the constructor of the param server. This helps us later if we decide to have different param servers, and decoupling is always a good idea

Reply:
I'm currently creating a global process group that contains both the "param server" and the trainer, which is why this is in one function, or it would hang forever while waiting for the last rank to join